app/models/smart_proxy.rb

```diff
@@ -19,10 +19,11 @@ class SmartProxy < ActiveRecord::Base
   has_many :puppet_ca_hosts, :class_name => 'Host::Managed',  :foreign_key => 'puppet_ca_proxy_id'
   has_many :puppet_ca_hostgroups, :class_name => 'Hostgroup', :foreign_key => 'puppet_ca_proxy_id'
   has_many :realms,                                           :foreign_key => 'realm_proxy_id'
-  URL_HOSTNAME_MATCH = %r{\A(?:http|https):\/\/([^:\/]+)}
   validates :name, :uniqueness => true, :presence => true
-  validates :url, :presence => true, :format => { :with => URL_HOSTNAME_MATCH, :message => N_('is invalid - only  http://, https:// are allowed') },
-            :uniqueness     => { :message => N_('Only one declaration of a proxy is allowed') }
+  validates :url, :presence => true,
+    :format => { :with => /\A#{URI.regexp(['http', 'https'])}\z/,
+                 :message => N_('is invalid - only  http://, https:// are allowed') },
```

Comment:
Two spaces between only and http here. Also, this error makes it sound like it's only checking the protocol. I'd maybe go with something like `is invalid - only http://, https:// are allowed and url must be valid`.